src/session.rs

```diff
@@ -32,12 +33,257 @@ pub(crate) enum Cmd {
+            }
+            WebDriverCommand::GetElementAttribute(ref we, ref attr) => {
+                base.join(&format!("element/{}/attribute/{}", we.0, attr))
+            }
+            WebDriverCommand::GetElementProperty(ref we, ref prop) => {
+                base.join(&format!("element/{}/property/{}", we.0, prop))
+            }
+            WebDriverCommand::GetCSSValue(ref we, ref attr) => {
+                base.join(&format!("element/{}/css/{}", we.0, attr))
+            }
+            WebDriverCommand::GetElementText(ref we) => {
+                base.join(&format!("element/{}/text", we.0))
+            }
+            WebDriverCommand::GetElementTagName(ref we) => {
+                base.join(&format!("element/{}/name", we.0))
+            }
+            WebDriverCommand::GetElementRect(ref we) => {
+                base.join(&format!("element/{}/rect", we.0))
+            }
+            WebDriverCommand::IsEnabled(ref we) => base.join(&format!("element/{}/enabled", we.0)),
+            WebDriverCommand::ExecuteScript(..) if self.is_legacy() => base.join("execute"),
+            WebDriverCommand::ExecuteScript(..) => base.join("execute/sync"),
+            WebDriverCommand::ExecuteAsyncScript(..) => base.join("execute/async"),
+            WebDriverCommand::GetCookies
+            | WebDriverCommand::AddCookie(_)
+            | WebDriverCommand::DeleteCookies => base.join("cookie"),
+            WebDriverCommand::GetNamedCookie(ref name)
+            | WebDriverCommand::DeleteCookie(ref name) => base.join(&format!("cookie/{}", name)),
+            WebDriverCommand::GetTimeouts | WebDriverCommand::SetTimeouts(..) => {
+                base.join("timeouts")
+            }
+            WebDriverCommand::ElementClick(ref we) => base.join(&format!("element/{}/click", we.0)),
+            WebDriverCommand::ElementClear(ref we) => base.join(&format!("element/{}/clear", we.0)),
+            WebDriverCommand::ElementSendKeys(ref we, _) => {
+                base.join(&format!("element/{}/value", we.0))
+            }
+            WebDriverCommand::PerformActions(..) | WebDriverCommand::ReleaseActions => {
+                base.join("actions")
+            }
+            WebDriverCommand::DismissAlert => base.join("alert/dismiss"),
+            WebDriverCommand::AcceptAlert => base.join("alert/accept"),
+            WebDriverCommand::GetAlertText | WebDriverCommand::SendAlertText(..) => {
+                base.join("alert/text")
+            }
+            WebDriverCommand::TakeScreenshot => base.join("screenshot"),
+            WebDriverCommand::TakeElementScreenshot(ref we) => {
+                base.join(&format!("element/{}/screenshot", we.0))
+            }
+            WebDriverCommand::Print(..) => base.join("print"),
+            WebDriverCommand::Status => unreachable!(),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn method_and_body(&self, request_url: &url::Url) -> (http::Method, Option<String>) {
```

Comment:
And same here?

Reply:
Yep this one was a straight copy/paste except for renaming `url` => `request_url` (just wanted to disambiguate vs the `url` crate in the global namespace) and also `self.is_legacy` => `self.is_legacy()`.